dapr/proto/components/v1/state.proto

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+syntax = "proto3";
+
+package dapr.proto.components.v1;
+
+import "google/protobuf/any.proto";
+import "dapr/proto/common/v1/common.proto";
+import "dapr/proto/components/v1/common.proto";
+
+option go_package = "github.com/dapr/dapr/pkg/proto/components/v1;components";
+
+
+// QueriableStateStore service provides a gRPC interface for querier state store components.
+service QueriableStateStore {
+  // Query performs a query request on the statestore.
+  rpc Query(QueryRequest) returns (QueryResponse) {}
+}
```

Comment:
I noticed we are missing `Ping` and `Init` both here and in the Transactional one. I guess the assumption is that one **must** implement a `service StateStore`  before exposing a `QueriableStateStore` or a `TransactionalStateStore`. Correct? Might be worth stating this in the comments.

Reply:
That's a good point thanks for asking this,

I got a bit confusing about this as too, but at the end of the day you need a `StateStore` implementation for either, queriable and transacional, you can see [here](https://github.com/dapr/components-contrib/blob/master/state/transactional_store.go#L17) and [here](https://github.com/dapr/components-contrib/blob/master/state/store.go#L98) that we do require init method for transacional state store but not for querier.

However I noticed that it does not make difference because we don't have a separate registry for the transactionalstatestore or for the querier. Looks like that statestore is the base interface for the others and in runtime - based on state store features, a [cast](https://github.com/dapr/dapr/blob/a6153bb26c5ae21dbb88569d7faf42d5011003f8/pkg/grpc/api.go#L293) to transactional state store is made. So it's not necessary to repeat ping/init methods for those.

Reply:
Might be worth adding as a comment this to the .proto, that one must define a base StateStore over which Queries and Transactions can be run.

Reply:
added!